runtime/src/Modules.tsx

```diff
@@ -330,13 +323,13 @@ const fetchPipeline = async (load: Load) => {
                 throw new Error(`Request failed with status ${res.status}: ${res.statusText}`);
               }
             } catch (e) {
-              if (url !== SNACKAGER_CDN_STAGING) {
+              if (urlIndex < SNACKAGER_API_URLS.length - 1) {
```

Comment:
getting little confused for this statement, could you try to add some comment for it?

Reply:
I just rewrote this part for clarity. It's now:

```ts
for (let i = 0; i < SNACKAGER_API_URLS.length; i++) {
  const url = SNACKAGER_API_URLS[i];
  const fetchFrom = `${url}/${handle}-${Platform.OS}/bundle.js`;

  // Check if we can retry with another URL, this would determine if we need to throw or swallow errors
  const hasFallbackUrl = i < SNACKAGER_API_URLS.length - 1;

  try {
    // Fetch dependency from snackager
  } catch (e) {
    if (!hasFallbackUrl) {
      // Abort, throw error
    } else {
      // Warn, but continue with next URL
    }
  }
```

Does that help?